src/rpc/dial.rs

```diff
@@ -72,6 +72,60 @@ impl RPCCredentials {
     }
 }
 
+impl ViamChannel {
+    async fn create_resp(
```

Comment:
note to reviewers: this is big chunk of code but it's basically just factoring out some code into a helper function. Shouldn't need too much look.